synapse/storage/schema/delta/53/user_share.sql

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2017 Vector Creations Ltd, 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- Old disused version of the tables below.
+DROP TABLE users_who_share_rooms;
```

Comment:
`DROP TABLE IF EXISTS` is preferable (likewise `create table if not exists`, `create index if not exists`) because occasionally you can end up with a delta being run twice.

(this tends to get forgotten, which is why a whole bunch of the other deltas don't have it, but it's supposed to be a thing.)

Reply:
There's also an argument that we should remove the `CREATE TABLE`s from the deltas that create them, since they're just pointless. I don't mind about that though.

Reply:
added the not exists/if exists/etc.

I think removing them from the deltas is maybe a good idea, but not something I'll do here.

Reply:
lmao i didnt do it for the indexes because thats a pg 9.5+ feature :)

Reply:
> lmao i didnt do it for the indexes because thats a pg 9.5+ feature :)

in theory you could `DROP INDEX IF EXISTS` it before creating it, but happy to leave this as-is.